client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -1,26 +1,60 @@
 /**
+ * External dependencies
+ */
+import { get, find, isObject } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import { getSelectedSiteWithFallback } from '../sites/selectors';
 
 export function getPromotions( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
-	const { promotions } = rootState.extensions.woocommerce.sites[ siteId ];
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
 	return promotions.promotions;
 }
 
+export function getPromotion( rootState, promotionId, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = getPromotions( rootState, siteId );
+	return ( find( promotions, ( p ) => promotionId === p.id ) || null );
+}
+
 export function getPromotionsPage( rootState, siteId = getSelectedSiteWithFallback( rootState ), page, perPage ) {
 	const offset = ( page - 1 ) * perPage;
 	const promotions = getPromotions( rootState, siteId );
 	return ( promotions ? promotions.slice( offset, ( offset + perPage ) ) : null );
 }
 
 export function getPromotionsCurrentPage( rootState ) {
-	const { promotions } = rootState.extensions.woocommerce.ui;
-	return promotions.currentPage;
+	const list = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'list' ], {} );
+	return list.currentPage;
 }
 
 export function getPromotionsPerPage( rootState ) {
-	const { promotions } = rootState.extensions.woocommerce.ui;
-	return promotions.perPage;
+	const list = get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'list' ], {} );
+	return list.perPage;
```

Comment:
Shouldn't this be something more like 

`return get( rootState, [ 'extensions', 'woocommerce', 'ui', 'promotions', 'list', 'perPage` ], null );`? Otherwise this could be undefined. Same comment above.

Reply:
Does it really matter if it returns `undefined` versus `null`?